Explorer/Assets/DCL/Notifications/NotificationsBus/NotificationsBusController.cs

```diff
@@ -6,34 +6,49 @@ namespace DCL.Notification.NotificationsBus
 {
     public class NotificationsBusController : INotificationsBusController
     {
-        public delegate void EventDelegate(params object[] parameters);
-        public event Action<INotification> OnNotificationAdded;
-        private Dictionary<NotificationType, EventDelegate> notificationSubscribers = new();
+        public delegate void NotificationClickedDelegate(params object[] parameters);
+        public delegate void NotificationReceivedDelegate(INotification notification);
+
+        private readonly Dictionary<NotificationType, NotificationClickedDelegate> notificationClickedSubscribers = new();
+        private readonly Dictionary<NotificationType, NotificationReceivedDelegate> notificationReceivedSubscribers = new();
 
         public void AddNotification(INotification notification)
         {
-            OnNotificationAdded?.Invoke(notification);
+            if (notificationReceivedSubscribers.TryGetValue(notification.Type, out NotificationReceivedDelegate thisEvent))
+                thisEvent.Invoke(notification);
         }
 
         public void ClickNotification(NotificationType notificationType, params object[] parameters)
         {
-            if (notificationSubscribers.TryGetValue(notificationType, out EventDelegate thisEvent))
-            {
+            if (notificationClickedSubscribers.TryGetValue(notificationType, out NotificationClickedDelegate thisEvent))
                 thisEvent.Invoke(parameters);
+        }
+
+        public void SubscribeToNotificationTypeClick(NotificationType desiredType, NotificationClickedDelegate listener)
+        {
+            if (notificationClickedSubscribers.TryGetValue(desiredType, out NotificationClickedDelegate thisEvent))
+            {
+                thisEvent += listener;
+                notificationClickedSubscribers[desiredType] = thisEvent;
+            }
+            else
+            {
+                thisEvent += listener;
+                notificationClickedSubscribers.Add(desiredType, thisEvent);
             }
         }
 
-        public void SubscribeToNotificationType(NotificationType desiredType, EventDelegate listener)
+        public void SubscribeToNotificationTypeReceived(NotificationType desiredType, NotificationReceivedDelegate listener)
         {
-            if (notificationSubscribers.TryGetValue(desiredType, out EventDelegate thisEvent))
+            if (notificationReceivedSubscribers.TryGetValue(desiredType, out NotificationReceivedDelegate thisEvent))
             {
                 thisEvent += listener;
-                notificationSubscribers[desiredType] = thisEvent;
+                notificationReceivedSubscribers[desiredType] = thisEvent;
             }
             else
             {
                 thisEvent += listener;
-                notificationSubscribers.Add(desiredType, thisEvent);
+                notificationReceivedSubscribers.Add(desiredType, thisEvent);
             }
```

Comment:
I think you could simplify a little bit the code (for this method and the previous in the same file), if you wanted, putting it similar to:

```
yourSubscribersDictionary.TryGetvalue(desiredType, out YourDelegateTypeBlahBlah thisEvent)
thisEvent += listener;
notificationReceivedSubscribers[desiredType] = thisEvent;
```
if it's true that `thisEvent` is not null when teh `TryGetvalue()` fails

Reply:
good point, done